tests/test_session.py

```diff
@@ -109,23 +106,38 @@ def simple_func(session: CoveSession, a_string: str) -> str:
     assert cove_output["Results"] == expected
 
 
-def test_session_result_error_handler(patch_boto3_client: MagicMock) -> None:
-    # Raise an exception instead of an expected response from boto3
-    patch_boto3_client.client.return_value.describe_account.side_effect = MagicMock(
-        side_effect=ClientError(
-            {"Error": {"Message": "broken!", "Code": "OhNo"}}, "describe_account"
+@pytest.mark.usefixtures("mock_session")
+def test_session_result_error_handler(
+    org_accounts: List[AccountTypeDef], mocker: MockerFixture
+) -> None:
+    # Raise an exception instead of an expected response from boto3.
+    # See "How to test ClientError with moto?".
+    # https://github.com/spulec/moto/issues/4453
+    _make_api_call = (
+        botocore.client.BaseClient._make_api_call  # type: ignore[attr-defined]
```

Comment:
I'm not sure why Mypy needs to be coerced into accepting this reference. The method is defined here.

https://github.com/boto/botocore/blob/e34b6a36ae0c1571bcb5fd7034e817d3b397e283/botocore/client.py#L684

Is there a better way to do this?

Reply:
From a quick investigate, this is the botocore-stubs library stubfile not containing _under methods - annoyingly without botocore-stubs present in env this typechecks fine. I haven't gone any further into that rabbit hole :)

Perhaps rather than injecting a mocked internal method we could just patch at boto3 client level? May require a custom moto org instantiation but that feels like the more "front door" approach

Reply:
Moto to the rescue again. The cleanest way to do it is to patch the moto backend to inject the error. See [2353be3](https://github.com/connelldave/botocove/pull/36/commits/2353be398b1cd999b9bf803391b9955e9cd7553f)!

(I tried patching the boto3 client and needed to use subclassing and `__getattr__` and even more type coercions. It was ugly.)